master/internal/trial.go

```diff
@@ -770,3 +774,31 @@ func (t *trial) maybeRestoreAllocation() (*model.Allocation, error) {
 		)
 	}
 }
+
+func (t *trial) checkResourcePoolRemainingCapacity() error {
+	launchWarnings, err := t.rm.ValidateResourcePoolAvailability(
+		&sproto.ValidateResourcePoolAvailabilityRequest{
+			Name:   t.config.Resources().ResourcePool(),
+			Slots:  t.config.Resources().SlotsPerTrial(),
+			TaskID: &t.taskID,
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("checking resource availability: %v", err.Error())
+	}
+	if len(launchWarnings) > 0 {
+		exitReason := fmt.Sprintf(
+			"task ID %v slots requested exceeds %v resource pool capacity",
+			t.taskID,
+			t.config.Resources().ResourcePool(),
+		)
+		logrus.Error(exitReason)
+
+		return t.transition(model.StateWithReason{
+			State:               model.ErrorState,
+			InformationalReason: exitReason,
+		})
```

Comment:
I think we can just log this and move on, we don't need to error

Reply:
The job stuck in queue without this. Output of `test_log_policy_exclude_node_single_agent`:
```
ERRO[2023-11-01T14:13:08-04:00] task ID 57.65b30b87-e141-41fe-8816-7e1ee1a0ff36 slots requested exceeds default resource pool capacity
DEBU[2023-11-01T14:13:08-04:00] starting new trial allocation                 allocation-id=57.65b30b87-e141-41fe-8816-7e1ee1a0ff36.2 component=trial experiment-id=57 job-id=f54aab51-d245-46f5-a200-c8830908fe46 task-id=57.65b30b87-e141-41fe-8816-7e1ee1a0ff36 task-type=TRIAL trial-id=57 trial-run-id=2
DEBU[2023-11-01T14:13:08-04:00] requestResources add allocation               experiment-id=57 job-id=f54aab51-d245-46f5-a200-c8830908fe46 task-id=57.65b30b87-e141-41fe-8816-7e1ee1a0ff36 task-type=TRIAL trial-id=57 trial-run-id=2
INFO[2023-11-01T14:13:08-04:00] resources are requested by Trial 57 (Experiment 57) (Allocation ID: 57.65b30b87-e141-41fe-8816-7e1ee1a0ff36.2)  actor-local-addr=default actor-system=master allocation-id=57.65b30b87-e141-41fe-8816-7e1ee1a0ff36.2 go-type=resourcePool resource-pool=default restore=false restoring=false
```

Reply:
oh my bad, I thought the above code handled this check

https://github.com/determined-ai/determined/blob/7772fb39d487b66f326ffeb963a111a213214915/master/internal/api_command.go#L122

https://docs.determined.ai/latest/reference/deploy/master-config-reference.html#launch-error

I think we should just return an error only if that launch-error config is set?